language/move-model/src/ty.rs

```diff
@@ -393,45 +393,49 @@ impl Type {
         }
     }
 
-    pub fn into_struct_tag(self, env: &GlobalEnv) -> Option<StructTag> {
+    /// Attempt to convert this type into a language_storage::TypeTag
+    pub fn into_struct_type(self, env: &GlobalEnv) -> Option<StructType> {
         use Type::*;
-        if self.is_open() {
-            None
-        } else {
             Some (
                 match self {
 		    Struct(mid, sid, ts) =>
-                        env.get_struct_tag(mid, sid, &ts)
+                        env.get_struct_type(mid, sid, &ts)
                             .expect("Invariant violation: struct type argument contains incomplete, tuple, reference, or spec type"),
 
                     _ => return None
 		}
 	    )
-        }
     }
 
-    /// Attempt to convert this type into a language_storage::TypeTag
-    pub fn into_type_tag(self, env: &GlobalEnv) -> Option<TypeTag> {
+    /// Attempt to convert this type into a language_storage::Type
+    pub fn into_type(self, env: &GlobalEnv) -> Option<MType> {
```

Comment:
I guess this function name is a bit odd, because `self` is of type `Type`. and `Type::into_type()` feels... weird. I would call it `into_storage_type` or sth else.

Reply:
Renamed it to `into_normalized_type`